src/support/modify.c

```diff
@@ -490,16 +487,22 @@ __wt_modify_reconstruct_from_upd_list(
          */
         WT_ASSERT(session, cbt->slot != UINT32_MAX);
 
-        ret = __wt_value_return_buf(cbt, cbt->ref, &upd_value->buf, &tw
-#ifdef HAVE_DIAGNOSTIC
-          ,
-          &is_ovfl_rm
-#endif
-        );
-        WT_ERR(ret);
+        ret = __wt_value_return_buf(cbt, cbt->ref, &upd_value->buf, &tw);
+
+        /*
+         * We race with checkpoint reconciliation removing the overflow items. Retry the read as the
+         * value should now be appended to the update chain by checkpoint reconciliation.
+         */
+        if (ovfl_retry && ret == WT_RESTART) {
+            ovfl_retry = false;
+            WT_STAT_CONN_DATA_INCR(session, txn_read_race_overflow_remove);
+            goto retry;
+        }
 
-        /* The base value cannot be a removed overflow value. */
-        WT_ASSERT(session, !is_ovfl_rm);
+        /* We should not read overflow removed after retry. */
+        WT_ASSERT(session, ovfl_retry || ret == 0);
```

Comment:
Is this correct? We could have a different `ret` value from `__wt_value_return_buf` that will be handled by the `WT_ERR` call just after.

Reply:
Yes, that will cause us to crash but that is ok.

Reply:
Fixed

Reply:
fixed